src/react/components/pages/projectSettings/projectForm.test.tsx

```diff
@@ -0,0 +1,223 @@
+import React from "react";
+import ProjectForm from "./projectForm";
+import { mount } from "enzyme";
+import { MockFactory } from "../../../../models/mockFactory";
+import TagColors from "../../common/tagsInput/tagColors.json";
+
+import { Provider } from "react-redux";
+import { BrowserRouter as Router } from "react-router-dom";
+import { IApplicationState } from "../../../../models/applicationState";
+import initialState from "../../../../redux/store/initialState";
+import createReduxStore from "../../../../redux/store/store";
+import ProjectSettingsPage from "./projectSettingsPage";
+
+describe("Project Form Component", () => {
+    let onSubmit: (value: any) => void;
+
+    const mockFactory = new MockFactory();
+    const project = mockFactory.project();
+    const connections = mockFactory.connections();
+    const defaultState: IApplicationState = initialState;
+    const store = createReduxStore(defaultState);
+
+    describe("Existing project", () => {
+        onSubmit = jest.fn();
+        let wrapper: any = null;
+
+        beforeEach(() => {
+            wrapper = mount(
+                <Provider store={store}>
```

Comment:
I don't believe the project form component itself has any direct dependencies on redux or router and you can likely remove these wrapping components.

Reply:
I got some interesting errors when I tried to run without it. Something about a `Link`. But I'll remove and get back to you with a more specific error message

Reply:
Update: Here's the error message I get: `    Invariant Violation: You should not use <Link> outside a <Router>`

Reply:
I'm guessing it's from the `Add Connection` button